### PR TITLE
feat: Add `capacity_rebalance` support for self-managed worker groups

### PR DESCRIPTION
**Pull request by LAKostis**
## Description

- See https://docs.aws.amazon.com/autoscaling/ec2/userguide/capacity-rebalance.html. We already have most functionality, only this parameter is missing

Resolves https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1124

### CODE REVIEW

File: workers_launch_template.tf [hunk 1]
```diff
@@ -97,6 +97,11 @@ resource "aws_autoscaling_group" "workers_launch_template" {
     "health_check_grace_period",
     local.workers_group_defaults["health_check_grace_period"]
   )
+  capacity_rebalance = lookup(
```
Comment (barryib) [inline workers_launch_template.tf:100]: Can you please do the same for ASG for workers (Launch Configuration). See workers.tf.

**Review by barryib — CHANGES_REQUESTED** [review]

File: workers_launch_template.tf (hunk 1 above)
Comment (LAKostis) [reply workers_launch_template.tf:99]: done

**Review by barryib — APPROVED** [review]
Thanks @LAKostis for your contribution.

**Comment by barryib** [comment]
Just released it in [v17.1.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/tag/v17.1.0)

**Pull request merged**